File: typescript/structs/LinkedList.ts
class LinkedNode<T> {
  private next: LinkedNode<T>;
  private data: T;

  constructor(data: T){
    this.data = data;
    this.next = null;
  }

  public getData(): T {
    return this.data;
  }

  public setData(data: T): void {
    this.data = data;
  }

  public getNext(): LinkedNode<T> {
    return this.next;
  }

  public setNext(next: LinkedNode<T>): void {
    this.next = next;
  }

}

export class LinkedList<T> {
  private head: LinkedNode<T>;

  constructor(){
    this.head = null;
  }

  public add(data: T): void {
    if(this.head == null){
      this.head = new LinkedNode<T>(data);
    } else {
      var temp = this.head;
      while(temp.getNext() != null){
        temp = temp.getNext();
      }
      temp.setNext(new LinkedNode<T>(data));
    }
  }

  public isEmpty(): boolean {
    return this.head == null;
  }

  public size(): number {
    var temp = this.head;
    var count = 0;
    while(temp != null){
      count++;
      temp = temp.getNext();
    }
    return count;
  }

  public get(index: number): any {
    if(index > this.size()-1 || index < 0 || this.isEmpty()){
      throw "index out of bounds";
    }
    if(index == 0){
      return this.head.getData();
    }
    var temp = this.head;
    var count = 0;
    while(temp.getNext() != null && count != index){
      temp =temp.getNext();
      count++;
    }
    if(count == index){
      return temp.getData();
    }
    return -1;
  }

  public contains(data: T): boolean {
    if(this.isEmpty()){
      return false;
    }
    if(this.head.getData() == data){
      return true;
    }
    var temp = this.head;
    while(temp != null){
      if(temp.getData() == data){
        return true;
      }
      temp = temp.getNext();
    }
    return false;
  }

  public remove(index: number): any {
    if(index > this.size() - 1 || index < 0 || this.isEmpty()){
      throw "Index out of bounds";
    }
    if(index == 0){
      var temp = this.head;
      if(this.head.getNext() != null){
        this.head = null;
        this.head = temp.getNext();
      } else {
        this.head = null;
      }
      return temp;
    }
    var temp = this.head;
    var count = 0;
    while(temp.getNext().getNext() != null && count + 1 != index){
      count++;
      temp = temp.getNext();
    }
    var next = temp.getNext();
    temp.setNext(null);
    temp.setNext(next.getNext());
    return temp;
  }

  public toString(): string {
    if(this.head == null){
      return "Empty";
    } 
    var str = "";
    var temp = this.head;
    while(temp != null){
      str += temp.getData();
      temp = temp.getNext();
      str += " ";
    }
    return str;
  }
}

File: python/structs/LinkedList.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

    def setNext(self, next):
        self.next = next

    def getNext(self):
        return self.next

    def setData(self, data):
        self.data = data

    def getData(self):
        return self.data

    def __str__(self):
      return self.data

class LinkedList:
    def __init__(self):
      self.head = None

    def __str__(self):
      if(self.isEmpty()):
        return "Empty"
      else:
        s = ""
        temp = self.head
        s += temp.getData()
        while(temp.getNext() != None):
          s += ", "
          temp = temp.getNext()
          s += temp.getData()
        return s
    
    def isEmpty(self):
      return self.head == None

    def add(self, element):
      if(self.head == None):
        self.head = Node(element)
      else:
        temp = self.head
        while(temp.getNext() != None):
          temp = temp.getNext()
        temp.setNext(Node(element))
    
    def contains(self, element):
      if(self.isEmpty()):
        return False
      if(self.head == element):
        return True
      temp = self.head
      while(temp.getNext() != None):
        if(temp.getData() == element):
          return True
        temp = temp.getNext()
      return False
    
    def size(self):
      count = 0
      temp = self.head
      while(temp != None):
        count += 1
        temp = temp.getNext()
      return count

    def get(self, index):
      if(index > self.size() or index < 0 or self.isEmpty()):
        raise IndexError
      if(index == 0):
        return self.head.getData()
      else:
        temp = self.head
        count = 0
        while(count != index and temp.getNext() != None):
          count += 1
          temp = temp.getNext()
        return temp.getData()
    
    def remove(self, index):
      if(index > self.size() or index < 0 or self.isEmpty()):
        raise IndexError
      if(index == 0):
        if(self.head.getNext() == None):
          temp = self.head
          self.head = None
          return temp.getData()
        else:
          temp = self.head
          self.head = None
          self.head = temp.getNext()
          return temp.getData()
      else:
        count = 0
        temp = self.head
        while(count + 1 != index and temp.getNext().getNext() != None):
          temp = temp.getNext()
          count += 1
        next = temp.getNext()
        temp.setNext(None)
        temp.setNext(next.getNext())
        return next.getData()

    def clear(self):
      while(not self.isEmpty()):
        self.remove(self.size() - 1)
          


